src/backend/cdb/cdbpartition.c

```diff
@@ -5228,12 +5228,14 @@ get_part_rule(Relation rel,
 
 		lc = list_head(l1);
 		prule2 = (PgPartRule *) lfirst(lc);
-		if (prule2 && prule2->topRule && prule2->topRule->children)
+		Assert(prule2);
+		if (prule2->topRule && prule2->topRule->children)
 			pNode = prule2->topRule->children;
 
 		lc = lnext(lc);
 
 		pid2 = (AlterPartitionId *) lfirst(lc);
+		Assert(pid2);
```

Comment:
Why is this check here?

Reply:
This calls `get_part_rule1()`, so this assert is redundant:
https://github.com/arenadata/gpdb/blob/215d8d8f9fec39300e4eb84977ade4fce040ebc1/src/backend/cdb/cdbpartition.c#L4873-L4877